api/views.py

```diff
@@ -144,6 +160,293 @@ def get_queryset(self):
+    """
+    Get real phone number records for the authenticated user.
+
+    The authenticated user must have a subscription that grants one of the
+    `SUBSCRIPTIONS_WITH_PHONE` capabilities.
+
+    Client must be authenticated, and these endpoints only return data that is
+    "owned" by the authenticated user.
+
+    """
+    http_method_names = ["get", "post", "patch"]
+    permission_classes = [permissions.IsAuthenticated, HasPhoneService]
+    serializer_class = RealPhoneSerializer
+    # TODO: this doesn't seem to e working?
+    throttle_classes = [RealPhoneRateThrottle]
+
+    def get_queryset(self):
+        return RealPhone.objects.filter(user=self.request.user)
+
+
+    def create(self, request):
+        """
+        Add real phone number to the authenticated user.
+
+        The "flow" to verify a real phone number is:
+        1. POST a number (Will text a verification code to the number)
+        2a. PATCH the verification code to the realphone/{id} endpoint
+        2b. POST the number and verification code together
+
+        The authenticated user must have a subscription that grants one of the
+        `SUBSCRIPTIONS_WITH_PHONE` capabilities.
+
+        The `number` field should be in [E.164][e164] format which includes a country
+        code. If the number is not in E.164 format, this endpoint will try to
+        create an E.164 number by prepending the country code of the client
+        making the request (i.e., from the `X-Client-Region` HTTP header).
+
+        If the `POST` does NOT include a `verification_code` and the number is
+        a valid (currently, US-based) number, this endpoint will text a
+        verification code to the number.
+
+        If the `POST` DOES include a `verification_code`, and the code matches
+        a code already sent to the number, this endpoint will set `verified` to
+        `True` for this number.
+
+        [e164]: https://en.wikipedia.org/wiki/E.164
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        valid_number, error = _validate_number(request)
+        if valid_number == None:
+            return response.Response(
+                status=error["status"], data=error["data"]
+            )
+        serializer.validated_data["number"] = valid_number.phone_number
+
+        # To make the POST API more flexible, if the request includes a
+        # verification_code value, look for any un-expired record that
+        # matches both the phone number and verification code and mark it
+        # verified.
+        verification_code = serializer.validated_data.get("verification_code")
+        if verification_code:
+            valid_record = RealPhone.objects.filter(
+                user=request.user,
+                number=serializer.validated_data["number"],
+                verification_code=verification_code,
+                verification_sent_date__gt=(
+                    datetime.now() -
```

Comment:
Suggestion: `datetime.now(timezone.utc)` for timezone-aware dates. Seeing warnings in server logs:

```
/Users/john/.pyenv/versions/3.9.11/lib/python3.9/site-packages/django/db/models/fields/__init__.py:1416: RuntimeWarning: DateTimeField RealPhone.verification_sent_date received a naive datetime (2022-06-27 20:29:43.103116) while time zone support is active.
  warnings.warn("DateTimeField %s received a naive datetime (%s)"
```